### PR TITLE
add session_token for temp s3 access

### PR DESCRIPTION
**Pull request by ramarnat**
This is completely untested, but will try and do a build tonight/tomorrow. Take a look @wuputah

### CODE REVIEW

**Comment by wuputah** [comment]
looks right to me.

**Comment by ramarnat** [comment]
unfortunately, does not seem to work. Get a bad request error. How can I turn on debugging to see the exact http call being made?

File: src/pgduckdb_options.cpp [hunk 1]
```diff
@@ -65,6 +65,9 @@ ReadDuckdbSecrets() {
 		if (!is_null_array[Anum_duckdb_secret_region - 1])
 			secret.region = DatumToString(datum_array[Anum_duckdb_secret_region - 1]);
 
+		if (!is_null_array[Anum_duckdb_secret_session_token - 1])
+			secret.region = DatumToString(datum_array[Anum_duckdb_secret_session_token - 1]);
```
Comment (rafaelbey) [inline src/pgduckdb_options.cpp:69]: I'm not fully familiar, but interested in this feature.  

This is setting the session token into region.  Likely a copy/paste from the lines above?

**Comment by ramarnat** [comment]
Looks like that fixed it. Thank you @rafaelbey

**Review by mkaruza — APPROVED** [review]
Looks straight forward feature PR.

**Pull request merged**